manager/integration/tests/test_basic.py

```diff
@@ -4588,21 +4593,34 @@ def test_backup_volume_restore_with_access_mode(set_random_backupstore, # NOQA
        => Validate the access mode should be the same as the specified
     """
     # Step 1
-    test_volume_name = generate_volume_name()
+    test_volume_name = "test-volume-" + str(random.randint(0, 9))
     client.create_volume(name=test_volume_name,
                          size=str(DEFAULT_VOLUME_SIZE * Gi),
                          numberOfReplicas=2,
                          accessMode=access_mode)
     wait_for_volume_creation(client, test_volume_name)
     volume = wait_for_volume_detached(client, test_volume_name)
-    volume.attach(hostId=common.get_self_host_id())
-    volume = common.wait_for_volume_healthy(client, test_volume_name)
+
+    # Since RWX volumes can only be manually attached in maintenance mode,
+    #  the volume was created by PV and PVC deployed.
+    if access_mode == "rwx":
+        test_pv_name = test_volume_name + "-pv"
+        test_pvc_name = test_volume_name + "-pvc"
+
+        create_pv_for_volume(client, core_api, volume, test_pv_name)
+        create_pvc_for_volume(client, core_api, volume, test_pvc_name)
+        deployment = make_deployment_with_pvc(
+            'deployment-multi-pods-test', test_pvc_name, replicas=2)
+        create_and_wait_deployment(get_apps_api_client(), deployment)
```

Comment:
Regarding `create_and_wait_deployment(get_apps_api_client(), deployment)`
Do we need a standard code format like [this](https://github.com/longhorn/longhorn-tests/blob/master/manager/integration/tests/test_ha.py#L2757-L2759)?

Reply:
Basically, we should try to use pytest fixture instead (the code in longhorn-test here is not quite good formatted, but I think it should be fine here.

Need a further effort in the future when doing refactoring.